### PR TITLE
Tracking devel

### PR DESCRIPTION
**Pull request by a-link**
Adds continuous pipeline execution and tracking functionality to RoboSherlock.

### CODE REVIEW

File: robosherlock/descriptors/annotators/tracker/KCFTrackingAnnotator.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+annotator:
+  name: KCFTrackingAnnotator
+  implementation: rs_KCFTrackingAnnotator
+parameters:
```
Comment (bbferka) [inline robosherlock/descriptors/annotators/tracker/KCFTrackingAnnotator.yaml]: remove;

File: robosherlock/descriptors/annotators/filter/ClosestHypothesisFilter.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+annotator:
+  description: Filters the single object hypothesis closest to the that of id OBJ_ID_TRACK
+    during the first process call of ClosestHypothesisFilter, taking potential movement
+    of that hypothesis into account.
+  implementation: rs_ClosestHypothesisFilter
+  name: ClosestHypothesisFilter
+parameters:
```
Comment (bbferka) [inline robosherlock/descriptors/annotators/filter/ClosestHypothesisFilter.yaml]: remove

File: robosherlock/config/config_kinect_robot_tracking.ini [hunk 3]
```diff
@@ -0,0 +1,20 @@
+[camera]
```
Comment (bbferka) [inline robosherlock/config/config_kinect_robot_tracking.ini]: delete this file. There are way too many config files already in the core; I guess it is only the filterBlurred images that was changed; we can put that in the documentation

File: robosherlock/src/flowcontrol/src/RSProcessManager.cpp [hunk 4]
```diff
@@ -170,9 +172,9 @@ bool RSProcessManager::executePipelineCallback(robosherlock_msgs::ExecutePipelin
   if(req.annotators.empty())
     return false;
 
-  std::vector<std::string> newPipelineOrder = req.annotators;
+  std::vector<std::string> new_pipeline_orders[0] = req.annotators;
```
Comment (bbferka) [inline robosherlock/src/flowcontrol/src/RSProcessManager.cpp]: this is a very weird thing you are doing here...mixing array declarations with std::vectors...what is this? You can probably survive with a simple vect of strings

File: robosherlock/src/flowcontrol/src/RSProcessManager.cpp [hunk 5]
```diff
@@ -287,6 +291,113 @@ bool RSProcessManager::handleQuery(std::string &request, std::vector<std::string
...
+
+        std::vector<rs::ObjectHypothesis> obj_hyps;
+        scene.identifiables.filter(obj_hyps);
+        rs::Size s = rs::create<rs::Size>(*tcas); // This is just a hack to get a simple integer (the object ID) into the cas.
+        s.height.set(obj_id);
+        outInfo(FG_GREEN << "SETTING OBJ_TO_TRACK");
+        sceneCas.setFS("OBJ_ID_TRACK", s);
+        outInfo("Found " << obj_hyps.size() << " objects");
+
+        visualizer_.setActiveAnnotators(new_pipeline_orders[1]);
```
Comment (bbferka) [inline robosherlock/src/flowcontrol/src/RSProcessManager.cpp]: what if new_pipeline_order is only of size 1? check size of vect before accessing;

File: robosherlock/src/queryanswering/src/QueryInterface.cpp [hunk 6]
```diff
@@ -135,6 +167,62 @@ bool QueryInterface::handleDetect(std::vector<std::string> &res)
   return true;
 }
 
+bool QueryInterface::handleTrack(std::vector<std::string> &res)
+{
+  //res.push_back("CollectionReader"); // Not required because is added manually
```
Comment (bbferka) [inline robosherlock/src/queryanswering/src/QueryInterface.cpp]: remove commented code

File: robosherlock/descriptors/analysis_engines/demo.yaml [hunk 7]
```diff
@@ -10,4 +10,4 @@ fixedflow:
   - PointCloudClusterExtractor
   - ClusterMerger
 CollectionReader:
-  camera_config_files: ['config_kinect_robot.ini']
\ No newline at end of file
+  camera_config_files: ['config_kinect_robot.ini']
```
Comment (a-link) [inline robosherlock/descriptors/analysis_engines/demo.yaml]: Is there some easy way to just get rid of this "change"?

**Pull request closed without merging**